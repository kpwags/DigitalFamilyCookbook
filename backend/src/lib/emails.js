const requestResetPasswordEmail = (name, resetToken) => `
  <div className="email" style="
    border: 1px solid black;
    padding: 20px;
    font-family: sans-serif;
    line-height: 2;
    font-size: 20px;">
      <h2>Hello ${name},</h2>
      <p>A request was made to reset your password. Please <a href="${
  process.env.DFC_FRONTEND_URL
}/reset-password?resetToken=${resetToken}">click here to reset your password</a> or copy the following link into your browser</p>
      <p>${process.env.DFC_FRONTEND_URL}/reset-password?resetToken=${resetToken}</p>
      <p>If you did not request a reset, please disregard and delete this email.</p>
      <p>Thanks,<br/>${process.env.DFC_SITE_TITLE}</p>
  </div>
`;

exports.requestResetPasswordEmail = requestResetPasswordEmail;
