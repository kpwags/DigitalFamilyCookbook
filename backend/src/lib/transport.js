const nodemailer = require('nodemailer');

const transport = nodemailer.createTransport({
  host: process.env.DFC_MAIL_HOST,
  port: process.env.DFC_MAIL_PORT,
  auth: {
    user: process.env.DFC_MAIL_USER,
    pass: process.env.DFC_MAIL_PASS,
  },
});

exports.transport = transport;
