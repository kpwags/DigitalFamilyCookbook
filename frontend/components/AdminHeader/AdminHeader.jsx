import PropTypes from 'prop-types';
import styled from 'styled-components';

const PageHeader = styled.div`
    width: 96%;
    margin: 0 2% 40px 2%;
    display: grid;
    grid-template-columns: 3fr 1fr;

    .title {
        grid-column-start: 1;
        grid-column-end: 1;
        justify-self: start;

        h1 {
            margin: 0;
            padding: 0;
        }
    }

    .add-button {
        grid-column-start: 2;
        grid-column-end: 2;
        justify-self: end;
    }

    @media all and (max-width: 800px) {
        grid-template-columns: 2fr 2fr;
    }
`;

const AdminHeader = props => {
    return (
        <PageHeader id="admin-header">
            <div className="title">
                <h1>{props.title}</h1>
            </div>
            <div className="add-button">{props.children}</div>
        </PageHeader>
    );
};

AdminHeader.propTypes = {
    title: PropTypes.string.isRequired,
    children: PropTypes.node
};

export { AdminHeader };
