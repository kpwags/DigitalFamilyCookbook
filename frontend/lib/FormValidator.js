import validator from 'validator';

class FormValidator {
    static validateEmail(value) {
        if (typeof value !== 'string' || !validator.isEmail(value)) {
            return false;
        }

        return true;
    }

    static validateInvitationCode(value) {
        if (!this.validateNotEmpty(value)) {
            return {
                valid: false,
                message: 'Invitation code is required'
            };
        }

        const regEx = new RegExp('^[A-Za-z0-9_]+$');
        if (!regEx.test(value)) {
            return {
                valid: false,
                message: 'Invitation codes can only contain letters, numbers, and underscores'
            };
        }

        if (value.trim().length < 1 || value.trim().length > 20) {
            return {
                valid: false,
                message: 'Invitation codes must be between 1 and 20 characters'
            };
        }

        return {
            valid: true,
            message: ''
        };
    }

    static validateNotEmpty(value) {
        if (typeof value !== 'string' || value.trim().length === 0) {
            return false;
        }

        return true;
    }

    static validateSetLength(value, fieldName = 'value', minLength = 1, maxLength = 25) {
        if (!this.validateNotEmpty(value)) {
            return {
                valid: false,
                message: `${fieldName} is required`
            };
        }

        if (value.trim().length < minLength || value.trim().length > maxLength) {
            return {
                valid: false,
                message: `${fieldName} must be between ${minLength} and ${maxLength} characters`
            };
        }

        return {
            valid: true,
            message: ''
        };
    }

    static validateUsername(value) {
        if (!this.validateNotEmpty(value)) {
            return {
                valid: false,
                message: 'Username is required'
            };
        }

        const regEx = new RegExp('^[A-Za-z0-9_]+$');
        if (!regEx.test(value)) {
            return {
                valid: false,
                message: 'Username can only contain letters, numbers, and underscores'
            };
        }

        if (value.trim().length < 1 || value.trim().length > 20) {
            return {
                valid: false,
                message: 'Username must be between 1 and 20 characters'
            };
        }

        return {
            valid: true,
            message: ''
        };
    }

    static validatePassword(value1, value2) {
        if (typeof value1 !== 'string' || typeof value2 !== 'string' || (value1 === '' && value2 === '')) {
            return {
                valid: false,
                message: 'Password is required'
            };
        }

        if (value1.trim() !== value2.trim()) {
            return {
                valid: false,
                message: 'Passwords do not match'
            };
        }

        if (value1.trim().length < 8) {
            return {
                valid: false,
                message: 'Passwords must be at least 8 characters long'
            };
        }

        return {
            valid: true,
            message: ''
        };
    }
}

export { FormValidator };
