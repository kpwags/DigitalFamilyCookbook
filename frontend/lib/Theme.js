const Theme = {
    black: 'hsl(0, 0%, 22.4%)',
    green: 'hsl(140, 100%, 25%)',
    lightGreen: 'hsl(136, 44%, 47%)',
    paleGreen: 'hsl(136, 100%, 88%)',
    darkGreen: 'hsl(140, 100%, 15%)',
    bs: '0 12px 24px 0 hsla(0, 0%, 0%, 0.2)'
};

export { Theme };
