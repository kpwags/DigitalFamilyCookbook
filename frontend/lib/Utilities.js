import Router from 'next/router';

class Utilities {
    static setTableRowActive(tableId, rowId) {
        const table = document.getElementById(tableId);
        const tbody = table.getElementsByTagName('tbody')[0];

        // TODO: Find better way?
        // eslint-disable-next-line no-restricted-syntax
        for (const row of tbody.getElementsByTagName('tr')) {
            row.className = '';
        }

        document.getElementById(rowId).className = 'selected';
    }

    // TODO: Add Message
    static invalidateField(elementId, errorMessage = '') {
        document.getElementById(elementId).className = 'errored';
        document.getElementById(elementId).parentElement.className = 'errored';

        if (errorMessage !== '') {
            document.getElementById(`${elementId}-message`).innerText = errorMessage;
            document.getElementById(`${elementId}-message`).style.display = 'block';
        }
    }

    static invalidateFieldByClass(className) {
        const elements = document.getElementsByClassName(className);

        for (let i = 0; i < elements.length; i += 1) {
            elements[i].className = `${className}  errored`;
        }
    }

    static resetFieldByClass(className) {
        const elements = document.getElementsByClassName(className);

        for (let i = 0; i < elements.length; i += 1) {
            let elementClassName = elements[i].className;
            elementClassName = elementClassName.replace('errored', '');
            elements[i].className = elementClassName;
        }
    }

    static resetField(elementId) {
        let { className } = document.getElementById(elementId);
        className = className.replace('errored', '');
        document.getElementById(elementId).className = className;

        let parentClassName = document.getElementById(elementId).parentElement.className;
        parentClassName = parentClassName.replace('errored', '');
        document.getElementById(elementId).parentElement.className = parentClassName;
        document.getElementById(`${elementId}-message`).style.display = 'none';
        document.getElementById(`${elementId}-message`).innerText = '';
    }

    static getNextAvailableValue(arr, idx = 'key') {
        let nextValue = 0;
        arr.forEach(item => {
            if (item[idx] >= nextValue) {
                nextValue = item[idx];
            }
        });

        nextValue += 1;

        return nextValue;
    }

    static convertCookingTime(time) {
        let timeHours = 0;
        let timeMinutes = 0;

        let returnString = '';

        if (time !== null && parseInt(time, 10) > 0) {
            timeHours = Math.floor(time / 60);
            timeMinutes = time % 60;

            if (timeHours === 1) {
                returnString += `${timeHours} hour`;
            } else if (timeHours > 1) {
                returnString += `${timeHours} hours`;
            }

            if (timeHours > 0 && timeMinutes > 0) {
                returnString += ', ';
            }

            if (timeMinutes > 0) {
                returnString += `${timeMinutes} minutes`;
            }
        }

        return returnString;
    }

    static convertIDArray(arr) {
        const newArr = [];
        arr.forEach(i => {
            newArr.push(i.id);
        });

        return newArr;
    }

    static toTitleCase(str) {
        return str
            .toLowerCase()
            .split(' ')
            .map(word => {
                return word.replace(word[0], word[0].toUpperCase());
            })
            .join(' ');
    }

    static goToRecipe(id) {
        Router.push({
            pathname: '/recipe',
            query: { id }
        });
    }

    static toggleElement(elementId) {
        if (document.getElementById(elementId).style.display !== 'none') {
            document.getElementById(elementId).style.display = 'none';
        } else {
            document.getElementById(elementId).style.display = 'block';
        }
    }

    static toggleMobileMenu() {
        document.getElementById('hamburgermenu').classList.toggle('change');

        if (document.getElementById('mobilemenu').style.left === '0px') {
            document.getElementById('mobilemenu').style.left = '-250px';
        } else {
            document.getElementById('mobilemenu').style.left = '0px';
        }
    }

    static toggleHeaderMenu(menuId) {
        if (
            document.getElementById(menuId).style.display === 'none' ||
            document.getElementById(menuId).style.display === ''
        ) {
            document.getElementById(menuId).style.display = 'block';
        } else {
            document.getElementById(menuId).style.display = 'none';
        }
    }

    static hideHeaderMenu(menuId) {
        document.getElementById(menuId).style.display = 'none';
    }

    static formatDate(date) {
        const formattedDate = new Date(date);
        return formattedDate.toLocaleDateString('en-US');
    }

    static boxColors = ['gray', 'blue', 'green', 'orange', 'purple'];
}

export { Utilities };
