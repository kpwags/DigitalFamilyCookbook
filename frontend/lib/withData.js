import withApollo from 'next-with-apollo';
import ApolloClient from 'apollo-boost';
import { endpoint } from '../config';
import { LOCAL_STATE_QUERY } from '../queries/Local';

function createClient({ headers }) {
    return new ApolloClient({
        uri: endpoint,
        request: operation => {
            operation.setContext({
                fetchOptions: {
                    credentials: 'include'
                },
                headers
            });
        },
        // local data
        clientState: {
            resolvers: {
                Mutation: {
                    toggleOverlay(_, variables, { cache }) {
                        const { overlayVisible } = cache.readQuery({
                            query: LOCAL_STATE_QUERY
                        });

                        const data = {
                            data: {
                                overlayVisible: !overlayVisible
                            }
                        };

                        cache.writeData(data);

                        return data;
                    }
                }
            },
            defaults: {
                overlayVisible: false
            }
        }
    });
}

export default withApollo(createClient);
