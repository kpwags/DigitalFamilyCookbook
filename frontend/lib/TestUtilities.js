/* eslint-disable import/no-extraneous-dependencies */
import casual from 'casual';
import PropTypes from 'prop-types';
import { MockedProvider } from '@apollo/react-testing';
import { ThemeProvider } from 'styled-components';
import { Theme } from './Theme';

const MockedThemeProvider = (props) => {
    const { mocks = null, addTypename = false, children } = props;
    return (
        <ThemeProvider theme={Theme}>
            <MockedProvider mocks={mocks} addTypename={addTypename}>
                {children}
            </MockedProvider>
        </ThemeProvider>
    );
};

MockedThemeProvider.propTypes = {
    children: PropTypes.node,
    mocks: PropTypes.array,
    addTypename: PropTypes.bool,
};

const TestAdmin = () => ({
    __typename: 'User',
    id: casual.uuid,
    name: casual.name,
    username: casual.username,
    email: casual.email,
    bio: 'I am just a test admin',
    image: 'user.jpg',
    largeImage: 'user-lg.jpg',
    permissions: ['USER', 'ADMIN'],
});

const TestCategory = () => ({
    __typename: 'Category',
    id: casual.uuid,
    name: casual.word,
    createdAt: casual.date,
});

const TestDirection = () => ({
    __typename: 'Direction',
    id: casual.uuid,
    name: casual.words(),
    sortOrder: casual.integer,
});

const TestIngredient = () => ({
    __typename: 'Ingredient',
    id: casual.uuid,
    name: casual.words(),
    sortOrder: casual.integer,
});

const TestInvitationCode = () => ({
    __typename: 'InvitationCode',
    id: casual.uuid,
    code: casual.word,
    createdAt: casual.date,
});

const TestMeat = () => ({
    __typename: 'Meat',
    id: casual.uuid,
    name: casual.words(),
    createdAt: casual.date,
});

const TestUser = () => ({
    __typename: 'User',
    id: casual.uuid,
    name: casual.name,
    username: casual.word,
    email: casual.email,
    bio: 'I am just a test user',
    image: 'user.jpg',
    largeImage: 'user-lg.jpg',
    permissions: ['USER'],
});

const TestRecipe = () => ({
    __typename: 'Recipe',
    id: casual.uuid,
    name: casual.name,
    public: casual.boolean,
    source: casual.words(),
    sourceUrl: casual.url,
    time: casual.integer,
    activeTime: casual.integer,
    servings: casual.integer,
    calories: casual.integer,
    carbohydrates: casual.integer,
    fat: casual.integer,
    sugar: casual.integer,
    cholesterol: casual.integer,
    fiber: casual.integer,
    image: 'recipe.jpg',
    largeImage: 'recipe-large.jpg',
    ingredients: [TestIngredient(), TestIngredient()],
    directions: [TestDirection(), TestDirection()],
    meats: [TestMeat(), TestMeat()],
    categories: [TestCategory(), TestCategory()],
    user: TestUser(),
});

const TestRecipesList = () => [TestRecipe(), TestRecipe(), TestRecipe(), TestRecipe(), TestRecipe()];

export { MockedThemeProvider, TestAdmin, TestCategory, TestDirection, TestIngredient, TestInvitationCode, TestMeat, TestRecipe, TestRecipesList, TestUser };
