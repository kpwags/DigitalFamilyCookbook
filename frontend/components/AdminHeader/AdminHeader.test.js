import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MockedThemeProvider } from '../../lib/TestUtilities';
import { AdminHeader } from './AdminHeader';

describe('<AdminHeader/>', () => {
    it('should match snapshot', () => {
        const wrapper = mount(
            <MockedThemeProvider>
                <AdminHeader title="Testing Admin Header" />
            </MockedThemeProvider>
        );

        expect(toJson(wrapper.find('div#admin-header'))).toMatchSnapshot();
    });
});
