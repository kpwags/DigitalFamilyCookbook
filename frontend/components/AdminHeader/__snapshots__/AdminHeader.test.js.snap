// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminHeader/> should match snapshot 1`] = `
<div
  className="AdminHeader__PageHeader-sc-1ikj811-0 ilNOFJ"
  id="admin-header"
>
  <div
    className="title"
  >
    <h1>
      Testing Admin Header
    </h1>
  </div>
  <div
    className="add-button"
  />
</div>
`;
